File: puzzles/day 2/python/two.one.py
from pathlib import Path
gamestrategy = []
score: int = 0
gamestrategy = Path("../input.txt").read_text("utf-8").split('\n', -1)


def gamelogic(round):
    opponentMove = round[0]
    strategicMove = round[1]
    if (opponentMove == "A"):
        if (strategicMove == "X"):
            return 3
        elif (strategicMove == "Y"):
            return 6
        else:
            return 0
    elif (opponentMove == "B"):
        if (strategicMove == "X"):
            return 0
        elif (strategicMove == "Y"):
            return 3
        else:
            return 6
    else:
        if (strategicMove == "X"):
            return 6
        elif (strategicMove == "Y"):
            return 0
        else:
            return 3


def moveScore(input: str):
    if (input == "X"):
        return 1
    elif (input == "Y"):
        return 2
    else:
        return 3


for round in gamestrategy:
    score += gamelogic(round.rstrip().lstrip().split(" "))
    score += moveScore(round.rstrip().lstrip().split(" "))
    print(moveScore(round.rstrip().lstrip().split(" ")[1]))

print("should be more than 13000")
print(score)

File: puzzles/day 2/js/two.one.js
const fs = require("fs");
let score =0;
fs.readFile('../input.txt', 'utf8',(err,data)=>{
    const arrayOfInput = data.split("\n");
    arrayOfInput.forEach(input => {
    //console.log(input)
    score += gamelogic(input)
    });
    console.log(score);
});


const gamelogic = (input)=> {
let inputarray = input.split("");
let P1 = inputarray[0];
let P2 = inputarray[2];
switch (P1) {
    case "A":
        if(P2 == "Y")return 8
        if(P2 == "X")return 4
        if(P2 == "Z")return 3
        break;
    case "B":
        if(P2 == "Y")return 5
        if(P2 == "X")return 1
        if(P2 == "Z")return 9
    break;

    case "C":
        if(P2 == "Y")return 2
        if(P2 == "X")return 7
        if(P2 == "Z")return 6
    break;
}
}
